### PR TITLE
Add a config to force enable lm and add more tests for lm

### LINKED ISSUE #5829: Support late materialization
## Feature Request

**Is your feature request related to a problem? Please describe:**

For some query like `select * from t where a + b <> 0;`, we now scan all columns and then do filter, which is heavy and we can do better.

**Describe the feature you'd like:**

Support late materialization. We can scan column(s) at first to filter out unnecessary data segment, and then just scan the rest columns.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #5829

Problem Summary:

### What is changed and how it works?

1. add a config `force_push_down_all_filters_to_scan`, default `false`, only used in tests
2. force enable lm for tests under `tests/fullstack-test`
3. minor refine some code of lm

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/docker/config/tiflash_dt_force_enable_lm.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 PingCAP, Ltd.
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tests/docker/config/tiflash_dt_force_enable_lm.toml]: fixed

**Review by hongyunyan — APPROVED** [review]
LGTM

**Review by hongyunyan — APPROVED** [review]
LGTM

**Pull request merged**